File: public/docs/_examples/architecture/ts/app/hero-list.component.ts
import { Component, OnInit }   from '@angular/core';

import { Hero }                from './hero';
import { HeroService }         from './hero.service';

// #docregion metadata
@Component({
  selector:    'hero-list',
  templateUrl: 'app/hero-list.component.html'
})
// #docregion class
export class HeroListComponent implements OnInit {
  // #enddocregion metadata
  heroes: Hero[];
  selectedHero: Hero;

  // #docregion ctor
  constructor(private service: HeroService) { }
  // #enddocregion ctor

  ngOnInit() {
    this.heroes = this.service.getHeroes();
  }

  selectHero(hero: Hero) { this.selectedHero = hero; }
  // #docregion metadata
}

File: public/docs/_examples/architecture/dart/lib/hero_list_component.dart
import 'package:angular2/core.dart';

import 'hero.dart';
import 'hero_detail_component.dart';
import 'hero_service.dart';

// #docregion metadata, providers
@Component(
    // #enddocregion providers
    selector: 'hero-list',
    templateUrl: 'hero_list_component.html',
    directives: const [HeroDetailComponent],
    // #docregion providers
    providers: const [HeroService])
// #docregion class
class HeroListComponent implements OnInit {
  // #enddocregion metadata, providers
  List<Hero> heroes;
  Hero selectedHero;
  // #docregion ctor
  final HeroService _heroService;

  HeroListComponent(this._heroService);
  // #enddocregion ctor

  void ngOnInit() {
    heroes = _heroService.getHeroes();
  }

  void selectHero(Hero hero) {
    selectedHero = hero;
  }
  // #docregion metadata, providers
}
